File: Board/src/main/java/com/example/board/domain/user/domain/repository/UserRepository.java
package com.example.board.domain.user.domain.repository;

import com.example.board.domain.user.domain.User;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;


@Repository
public interface UserRepository extends CrudRepository<User, Long> {
    Optional<User> findByAccountId(String accountId); //Optional : 예상치못한 NPE를 제공되는 메소드로 null값으로 인해 발생하는 예외를 처리
    Boolean existsByAccountId(String accountId);
}

File: Kopring/Dise_V2/src/main/kotlin/com/example/dise_v2/domain/user/domain/repository/UserRepository.kt
package com.example.dise_v2.domain.user.domain.repository

import com.example.dise_v2.domain.user.domain.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User, Long> {
    fun findByAccountId(accountId: String): User?
    fun existsByAccountId(accountd: String): Boolean
}